service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -0,0 +1,47 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Observation;
+
+import com.github.mustachejava.Mustache;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Component
+public class ObservationStatementMapper {
+
+    private static final Mustache OBSERVATION_STATEMENT_EFFECTIVE_TIME_TEMPLATE =
+        TemplateUtils.loadTemplate("ehr_observation_statement_effective_time_template.mustache");
+    private static final Mustache OBSERVATION_STATEMENT_EFFECTIVE_PERIOD_TEMPLATE =
+        TemplateUtils.loadTemplate("ehr_observation_statement_effective_period_template.mustache");
+
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    public String mapObservationToObservationStatement(Observation observation, boolean isNested) {
+        var observationStatementTemplateParameters = ObservationStatementTemplateParameters.builder()
+            .observationStatementId(randomIdGeneratorService.createNewId())
+            .comment(observation.getComment())
+            .issued(DateFormatUtil.formatDate(observation.getIssued()))
+            .isNested(isNested)
+            .build();
+
+        if (observation.hasEffectiveDateTimeType() && observation.getEffectiveDateTimeType().hasValue()) {
```

Comment:
3 returns in one function is not the best pattern, what do you think about changing it into one return and only configure it in the ifs? 

```
template = OBSERVATION_STATEMENT_EFFECTIVE_PERIOD_TEMPLATE
...

return  TemplateUtils.fillTemplate(template, observationStatementTemplateParameters);
````

but if you applied my previous comment this code would probably change